stix_shifter_modules/aws_athena/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,492 @@
+import re
+import json
+import os.path as path
+from datetime import datetime, timedelta
+try:
+    # 3.8 and up
+    from collections.abc import Iterable
+except ImportError:
+    from collections import Iterable
+
+TIMESTAMP_PATTERN = r"\d{4}(-\d{2}){2}T\d{2}(:\d{2}){2}(\.\d+)?Z"
+PROTOCOL_LOOKUP_JSON_FILE = 'json/network_protocol_map.json'
+GUARDDUTY_CONFIG = 'json/guardduty_config.json'
+
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "AND",
+        ComparisonExpressionOperators.Or: "OR",
+        ComparisonComparators.GreaterThan: ">",
+        ComparisonComparators.GreaterThanOrEqual: ">=",
+        ComparisonComparators.LessThan: "<",
+        ComparisonComparators.LessThanOrEqual: "<=",
+        ComparisonComparators.Equal: "=",
+        ComparisonComparators.NotEqual: "!=",
+        ComparisonComparators.Like: "LIKE",
+        ComparisonComparators.In: "IN",
+        ComparisonComparators.Matches: 'REGEXP_LIKE',
+        ObservationOperators.Or: 'UNION',
+        ObservationOperators.And: 'INTERSECT'
+    }
+
+    def __init__(self, pattern: Pattern, data_model_mapper, time_range):
+        self.dmm = data_model_mapper
+        self._time_range = time_range
+        self.service_type = self.dmm.dialect
+        self._protocol_lookup_needed = True if self.service_type in ['vpcflow'] else False
+        self._epoch_time = True if self.service_type in ['vpcflow'] else False
+        self.qualifier_string = ''
+        self.translated = self.parse_expression(pattern)
+
+    @staticmethod
+    def _format_set(values) -> tuple:
+        """
+        Formatting list of values in the event of IN operation
+        :param values: str
+        :return: list
+        """
+        if not isinstance(values, tuple):
+            values = values.element_iterator()
+        return tuple("{}".format(value) for value in values)
+
+    @staticmethod
+    def _format_match(value) -> str:
+        """
+        Formatting value in the event of MATCHES operation
+        :param value: str
+        :return: str
+        """
+        return '{}'.format(value)
+
+    @staticmethod
+    def _format_equality(value) -> str:
+        """
+        Formatting value in the event of EQUALS operation
+        :param value: str
+        :return: str
+        """
+        return "{}".format(value)
+
+    @staticmethod
+    def _format_like(value) -> str:
+        """
+        Formatting value in the event of LIKE operation
+        :param value: str
+        :return: str
+        """
+        value = "{value}".format(value=value)
+        return QueryStringPatternTranslator._escape_value(value)
+
+    @staticmethod
+    def _escape_value(value) -> str:
+        if isinstance(value, str):
+            return '{}'.format(value.replace('\\', '\\\\').replace('\"', '\\"').replace('(', '\\(').replace(')', '\\)'))
```

Comment:
this escaping of backslashes doesn't seem correct. single` \ `are translated into `\\\\` . I think this function is a copy of some old query constructors. have you look at the recent update of the same function in qradar: 
https://github.com/opencybersecurityalliance/stix-shifter/blob/cc23b8bae276859d562b23d5eb71f6703075ee5e/stix_shifter_modules/qradar/stix_translation/query_constructor.py#L75

Reply:
@mdazam1942 The escaping of single backslash is now removed and the change is updated to the PR branch.